### PR TITLE
MULE-19516: Support new SDK API error package 

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/JavaErrorParsingTestCase.java [hunk 1]
```diff
@@ -170,8 +165,9 @@ public void operationsWithErrorTypesButExtensionWithNone() throws Exception {
 
 
   @Extension(name = "Heisenberg")
-  @Operations({HeisenbergOperations.class, MoneyLaunderingOperation.class,
-      KillingOperations.class, HeisenbergScopes.class, HeisenbergRouters.class})
+  // @Operations({HeisenbergOperations.class, MoneyLaunderingOperation.class,
+  // KillingOperations.class, HeisenbergScopes.class, HeisenbergRouters.class})
+  @Operations(OperationWithThrows.class)
```
Comment (Bardammu) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/JavaErrorParsingTestCase.java:170]: Commented code

Comment (marianogonzalez) [reply modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/JavaErrorParsingTestCase.java:170]: removing the commented code, but removing the operations from this test extension is still the right thing to do

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/error/SdkrrorTypeProviderAdapter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+   */
+  public static ErrorTypeProvider from(Object value) {
+    checkArgument(value != null, "Cannot adapt null value");
+
+    if (value instanceof ErrorTypeProvider) {
+      return (ErrorTypeProvider) value;
+    } else if (value instanceof org.mule.runtime.extension.api.annotation.error.ErrorTypeProvider) {
+      return new SdkrrorTypeProviderAdapter((org.mule.runtime.extension.api.annotation.error.ErrorTypeProvider) value);
+    } else {
+      throw new IllegalArgumentException(String.format("Value of class '%s' is neither a '%s' or a '%s'",
```
Comment (Bardammu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/error/SdkrrorTypeProviderAdapter.java:44]: I think the text should be `neither a '%s' nor a '%s'. Also you can import the static String.format

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/error/SdkErrorTypeDefinitionAdapter.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+   * @throws IllegalArgumentException if the value is not an instance of an adaptable type
+   */
+  public static <E extends Enum<E>> ErrorTypeDefinition<E> from(Object value) {
+    checkArgument(value != null, "Cannot adapt null value");
+    if (value instanceof ErrorTypeDefinition) {
+      return (ErrorTypeDefinition<E>) value;
+    } else if (value instanceof org.mule.runtime.extension.api.error.ErrorTypeDefinition) {
+      return new SdkErrorTypeDefinitionAdapter<E>((org.mule.runtime.extension.api.error.ErrorTypeDefinition) value);
+    } else {
+      throw new IllegalArgumentException(String.format("Value of class '%s' is neither a '%s' or a '%s'",
```
Comment (Bardammu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/error/SdkErrorTypeDefinitionAdapter.java:43]: nor?

**Review by Bardammu — DISMISSED** [review]

**Pull request merged**